src/shared/features/AppletSettings/DeleteAppletSetting/DeleteAppletSetting.test.tsx

```diff
@@ -0,0 +1,81 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+import { fireEvent, screen, waitFor } from '@testing-library/react';
+import mockAxios from 'jest-mock-axios';
+
+import { SettingParam, renderWithProviders } from 'shared/utils';
+
+import { initialStateData } from 'redux/modules';
+import { page } from 'resources';
+import { Roles } from 'shared/consts';
+import * as encryptionFunctions from 'shared/utils/encryption';
+import { mockedAppletId, mockedCurrentWorkspace, mockedApplet, mockedPassword } from 'shared/mock';
+
+import { DeleteAppletSetting } from './DeleteAppletSetting';
+
+const route = `/dashboard/${mockedAppletId}/settings/${SettingParam.DeleteApplet}`;
+const routePath = page.appletSettingsItem;
+const preloadedState = {
+  workspaces: {
+    workspaces: initialStateData,
+    currentWorkspace: {
+      ...initialStateData,
+      ...mockedCurrentWorkspace,
+    },
+    roles: {
+      ...initialStateData,
+      data: {
+        [mockedAppletId]: [Roles.Manager],
+      },
+    },
+    applet: mockedApplet,
+    workspacesRoles: initialStateData,
+  },
+  applet: {
+    applet: {
+      ...initialStateData,
+      data: mockedApplet,
+    },
+  },
+  popups: {
+    data: {
+      applet: mockedApplet,
+      encryption: null,
+      popupProps: undefined,
+      deletePopupVisible: true,
+      duplicatePopupsVisible: false,
+      transferOwnershipPopupVisible: false,
+      publishConcealPopupVisible: false,
+    },
+  },
+};
+const getPublicKeyMock = () => Buffer.from(JSON.parse(mockedApplet?.encryption?.publicKey || ''));
+
+const mockedUseNavigate = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockedUseNavigate,
+}));
+
+describe('DeleteAppletSetting', () => {
+  test('should render and submit', async () => {
+    mockAxios.delete.mockResolvedValueOnce(null);
+    jest.spyOn(encryptionFunctions, 'getAppletEncryptionInfo').mockImplementation(() => ({
+      getPublicKey: getPublicKeyMock,
+    }));
+    renderWithProviders(<DeleteAppletSetting />, { preloadedState, route, routePath });
+
+    expect(screen.getByTestId('applet-settings-delete-applet-delete-button')).toBeVisible();
+    expect(screen.getByTestId('applet-settings-delete-applet-delete-password-popup')).toBeVisible();
+
+    fireEvent.change(screen.getByLabelText(/Password/), {
```

Comment:
why not used `userEvent.type(screen.getByLabelText(/Password/), mockedPassword)`?

[@testing-library/user-event](https://github.com/testing-library/user-event) is a package that's built on top of fireEvent, but it provides several methods that resemble the user interactions more closely. In the example above, fireEvent.change will simply trigger a single change event on the input. However the type call, will trigger keyDown, keyPress, and keyUp events for each character as well. It's much closer to the user's actual interactions. This has the benefit of working well with libraries that you may use which don't actually listen for the change event.